File: CS201/Replit/MergeLists.java
public ListNode weave(ListNode a, ListNode b) {
    ListNode result = new ListNode(a.info);
    ListNode curr = result;
    a = a.next;
    while (a != null && b != null) {
        curr.next = new ListNode(b.info);
        b = b.next;
        curr = curr.next;
        curr.next = new ListNode(a.info);
        a = a.next;
        curr = curr.next;
    }
    if (b != null) {
        curr.next = b;
    }
    return result;
}

File: CS201/StarCoder/MergeLists.py
'''
Write method weave that takes two lists with the same number of nodes and weaves them together alternating one node from list a and one node from list b, returning the weaved-together list.
The weaved-list always starts with the first node/value from a, then chooses the next value from list b, and alternates, choosing successive values alternatively from each list. The last value in the returned list will always be the last value from b.

For example, if a = [1,3,5,7] and b = [2,4,6,8], then the call weave(a,b) returns [1,2,3,4,5,6,7,8].

You can create new nodes are weave together existing nodes.

The ListNode class will be accessible when your method is tested.


public class ListNode {
    int info;
    ListNode next;
    ListNode(int x){
        info = x;
    }
    ListNode(int x, ListNode node){
        info = x;
	next = node;
    }
}
'''

def weave(a, b):
    # Your code here
    if len(a) == 0:
        return b
    elif len(b) == 0:
        return a
    else:
        result = []
        for i in range(len(a)):
            if i % 2 == 0:
                result.append(a[i])
            else:
                result.append(b[i])
        return result
